aws-java-sdk-s3/src/main/java/com/amazonaws/services/s3/transfer/internal/DownloadImpl.java

```diff
@@ -77,7 +74,7 @@ public synchronized ObjectMetadata getObjectMetadata() {
         if (s3Object != null) {
             return s3Object.getObjectMetadata();
         }
-        return objectMetadata;
+        return null;
```

Comment:
No, they wont. Null will only be returned to the customers who will be using the new API where object metadata is not retrieved before object is downloaded. After that even while using new API, object metadata will be populated properly. It's definitely not a breaking change.